src/server/scripts/Northrend/Ulduar/Ulduar/boss_assembly_of_iron.cpp

```diff
@@ -626,10 +626,16 @@ class boss_stormcaller_brundir : public CreatureScript
                             {
                                 float x = float(irand(-25, 25));
                                 float y = float(irand(-25, 25));
-                                me->GetMotionMaster()->MovePoint(0, me->GetPositionX() + x, me->GetPositionY() + y, me->GetPositionZ());
+
+                                Position pos = *me;
+
+                                pos.m_positionX += x;
+                                pos.m_positionY += y;
+
+                                me->GetMotionMaster()->MovePoint(0, pos);
                                 // Prevention to go outside the room or into the walls
                                 if (Creature* trigger = me->FindNearestCreature(NPC_WORLD_TRIGGER, 100.0f, true))
-                                    if (me->GetDistance(trigger) >= 50.0f)
+                                    if (pos.GetExactDist2d(trigger) >= 50.0f)
```

Comment:
could you describe this change more in detail ?

Reply:
if the calculated Stormcaller Brundir position is more than 50 yards away, move back to the trigger